src/ray/raylet/node_manager.cc

```diff
@@ -502,8 +502,18 @@ void NodeManager::PublishActorStateTransition(
     // RECONSTRUCTING or DEAD entries have an odd index.
     log_length += 1;
   }
+  // If we were successful to append a record to the GCS table that the actor
+  // has died, signal this to anyone receiving signals from this actor.
+  auto success_callback = [](gcs::AsyncGcsClient *client, const ActorID &id,
+                          const ActorTableDataT &data) {
+    auto redis_context = client->primary_context();
+    if (data.state == ActorState::DEAD || data.state == ActorState::RECONSTRUCTING) {
+      std::vector<std::string> args = {"XADD", id.hex(), "*", "signal", "ACTOR_DIED"};
+      RAY_CHECK_OK(redis_context->RunArgvAsync(args));
```

Comment:
Maybe wrap this as a `SendSignal` function in `AsyncGcsClient`?

Reply:
Good idea. However, let's do this when . we add more signal, and leave it this way for MVP.